### PR TITLE
improve/add C snippets for control, loops, functions and types

### PR DESCRIPTION
**Pull request by REmerald**
Improve formatting for these snippets; add some useful snippets.

### CODE REVIEW

**Comment by OkelleyDevelopment** [comment]
Hey @REmerald , I merged the other PR that resolve the issue so the snippets function for others while I review this PR. 

Note: You can resolve the conflict by removing the `"",` that you'd added in that last commit. If you wished for the newlines to be present, then simply add `\n`, ideally I want to really minimize the "dead space" used for stuff like `"",` especially on the new PRs we accept. 

Once that's resolved I can give this the approval and get this merged :sunglasses:

**Comment by REmerald** [comment]
Removed `"",` for now, maybe I'll add `\n` later in my next pull requests.

File: snippets/c.json [hunk 1]
```diff
@@ -130,118 +130,325 @@
...
-    "prefix": "clist",
+  "enum type": {
+    "prefix": "enumt",
     "body": [
-      "typedef struct _node * Link;",
-      "typedef struct _node node;",
-      "struct _node {",
-      "\tint value;",
-      "\tLink next;",
+      "typedef enum $1 ${1:Numbers};"
```
Comment (OkelleyDevelopment) [inline snippets/c.json:446]: Need a comma here

File: snippets/c.json [hunk 2]
```diff
@@ -130,118 +130,325 @@
...
+      "union ${1:MyUnion} {",
+      "\t$0",
+      "};"
+    ],
+    "description": "Code snippet for union"
+  },
+  "union type": {
+    "prefix": "uot",
+    "body": [
+      "typedef union $1 ${1:Cell};"
```
Comment (OkelleyDevelopment) [inline snippets/c.json:422]: Add a comma here

File: snippets/c.json [hunk 3]
```diff
@@ -130,118 +130,325 @@
...
     "description": "Code snippet for struct"
   },
-  "typedef struct": {
-    "prefix": "structt",
-    "body": ["typedef struct {", "\t$0", "} ${1:MyStruct};"],
-    "description": "Code snippet to define a type with struct"
+  "struct type": {
+    "prefix": "stt",
+    "body": [
+      "typedef struct $1 ${1:Box};"
```
Comment (OkelleyDevelopment) [inline snippets/c.json:403]: Add a comma here too

**Review by OkelleyDevelopment — CHANGES_REQUESTED** [review]
Thank you for the last update, everything still looks good except for the 3 spots missing commas. 

I'm watching this PR tonight, so if you get these changes made in the next few hours then I will review asap so we can close this out :sunglasses:

File: snippets/c.json (hunk 3 above)
Comment (REmerald) [reply snippets/c.json]: fixed

File: snippets/c.json (hunk 2 above)
Comment (REmerald) [reply snippets/c.json]: fixed

File: snippets/c.json (hunk 1 above)
Comment (REmerald) [reply snippets/c.json]: fixed

**Review by OkelleyDevelopment — APPROVED** [review]
I believe these are acceptable, good work :sunglasses:

**Pull request merged**